[Ingest Manager] Enforce license level for custom registry URL

jen-huang opened the pull request:
## Summary

This PR adjusts the license level required to use custom registry URL.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78920/)
* Commit: 59e1d395bb729508604557ca6f83005215ebb7d2

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/59e1d395bb729508604557ca6f83005215ebb7d2) | [after](https://github.com/elastic/kibana/commit/12d01250249801d232bb01234c3044ef2fd6e642) | diff |
| --- | --- | --- | --- |
| `ingestManager` | 510.2KB | 510.7KB | +542.0B |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78920/","number":"78920","commit":"59e1d395bb729508604557ca6f83005215ebb7d2"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

ph reviewed (approved):
changes LGTM, fyi @mostlyjason

ph commented:
cc @ruflin

mostlyjason commented:
LGTM thanks @jen-huang

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79750/)
* Commit: d6267933d0befe973a943eb68a367c5e82df7936

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### page load bundle size
| id | [before](https://github.com/elastic/kibana/commit/d6267933d0befe973a943eb68a367c5e82df7936) | [after](https://github.com/elastic/kibana/commit/28278abdda7b912b695f6d0db2d6471a8b9260e5) | diff |
| --- | --- | --- | --- |
| `ingestManager` | 540.9KB | 541.5KB | +542.0B |

### History
* :green_heart: [Build #78920](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78920/) succeeded 59e1d395bb729508604557ca6f83005215ebb7d2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78920/","number":"78920","commit":"59e1d395bb729508604557ca6f83005215ebb7d2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79750/","number":"79750","commit":"d6267933d0befe973a943eb68a367c5e82df7936"}]}
PIPELINE-->

ruflin commented:
For awareness, this is still an unsupported feature, see https://github.com/elastic/kibana/issues/77333 for more details.

@jen-huang Do we have any experimental message or similar attached to it?

jen-huang commented:
> @jen-huang Do we have any experimental message or similar attached to it?

@mostlyjason had similar concerns, it's undocumented at this point, in this PR I added some messaging in the _dev_ docs and added a code comment. Does that suffice?

ruflin commented:
@jen-huang I would go one step further and also have a log message on info level that it is experimental when used as I guess most users will not look at the code itself.

jfsiii commented:
@ruflin @jen-huang  updated that existing message as well. See https://github.com/jen-huang/kibana/blob/d6267933d0befe973a943eb68a367c5e82df7936/x-pack/plugins/ingest_manager/server/services/epm/registry/registry_url.ts#L43

ruflin commented:
@jfsiii This message is about the license level not that it is experimental?

jen-huang commented:
PR to emit logging when custom registry URL is used: https://github.com/elastic/kibana/pull/80768
